contracts/core/native_fallback/L1/Registry.sol

```diff
@@ -67,7 +66,24 @@ contract Registry is IRegistry, Ownable, Pausable, AccessControl {
      * @param _chainID The chain ID to check
      */
     modifier isRevocable(uint256 _chainID) {
-        require(BitMaps.get(_irrevocableChainIDBitmap, _chainID) == false, "ChainID is irrevocable");
+        require(
+            BitMaps.get(_irrevocableChainIDBitmap, uint256(_getChainRole(_chainID, true))) == false,
+            "ChainID is irrevocable"
+        );
+
+        _;
+    }
+
+    /**
+     * @dev Modifier to ensure the caller is a grantee for a specific chain ID
+     * @param _chainID The chain ID to check
+     * @notice This modifier checks if the caller is a grantee or an irrevocable grantee for the specified chain ID.
+     * If the caller is not authorized, it reverts with "Not authorized".
+     */
+    modifier canConfigChain(uint256 _chainID) {
+        require(
+            _isGrantee(msg.sender, _chainID) || _isIrrevocableGrantee(msg.sender, _chainID), "Registry: Not authorized"
```

Comment:
Now checks whether either the irrevocable or the revocable role has already been granted